src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java

```diff
@@ -307,13 +312,24 @@ public void deleteToken(@RequestBody Token token) throws AuthenticationException
                     // In case where user has more than two tokens, we apply the expiration of the longest token
                     // to the second longest token.
                     Instant timestamp = token.getExpiration();
-                    tokenService.delete(token.getId());
-                    tokens = tokenService.findByUser(token.getUser());
+                    Long deleteTokenId = token.getId();
+                    tokens = tokens.stream().filter(t -> !t.getId().equals(deleteTokenId)).collect(Collectors.toCollection(ArrayList::new));
                     Token longestToken = tokens.stream().max(Comparator.comparing(Token::getExpiration)).get();
                     if (timestamp.isAfter(longestToken.getExpiration())) {
                         longestToken.setExpiration(timestamp);
                         tokenService.save(longestToken);
                     }
+
+                    // Find the token stats for the token to be deleted and associate those stats with the longest token.
+                    tokenStatsService
+                        .getAllTokenStatsByTokenId(deleteTokenId)
```

Comment:
Could you update the token id on sql level instead of getting all tokens stats into cache?